combatEngine

```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
 };
 
 function getInitiative(){//Helper Function to create an initiative order.
+    var initiativeOrder = [];
     var needInitiative = fighters
-    var iterate = 0    
-    while (needInitiative.length > 0){}
+    //Finds the highest initiative of the unassigned units and moves it to the initiativeOrder array, repeats until no more need Initiative, then returns that array. Ties go to the unit earlier in the list.
+    while(needInitiative.length > 0){
+        if(wantLogs = true){
+            console.log("need Initiative is length " + needInitiative.length + " and contains: ")
+            console.log (needInitiative)
+        };
+        topInit = -999
+        for(const i in needInitiative){
+            if(wantLogs == true){
+                console.log("i is " + i);
+                console.log("current i's speed is: " + needInitiative[i].speed);
+            };
+            if (needInitiative[i].speed > topInit){
+                topInit = fighters[i].speed
+                topInitIndex = i
+            };
+        };
+        initiativeOrder.push(needInitiative.splice(topInitIndex,1)) 
+    };
+    if(wantLogs = true){
+        console.log("Initiative Order Is:")
+    console.log(initiativeOrder);
+    };
+    return initiativeOrder; 
 };
-        
+
 function haveBattle(){
     var turncount = 0
     while (fighters[0].hp > 0 && fighters[1].hp > 0){
```

Comment:
Avoid hard codes like this!
maybe something like...    while(fighters.filter(x => x.hp > 0).length > 0){ ...

Reply:
TIL the filter method. Hold please.

Reply:
Also off by 1, otherwise looks good.  We want to fight until there's 1 left, not nothing left.